### PR TITLE
Activate RDBCoveringIterator

### PR DESCRIPTION
**Pull request by neunhoef**
### Scope & Purpose

This PR fixes RDBCoveringIterator. So far, it has never worked, since it was never activated. This PR
also activates TDBCoveringIterator under suitable circumstances.

This is the backport to 3.10.

Bugs found and fixed:

1. endless loop because _gotIntervals was always false, in general just typo, fixed
2. endless loop because _scanInterval was not advanced
3. endless loop because `break` did not set _scanInternal to the end.

- [*] :hankey: Bugfix

### Checklist

- [*] Tests
  - [*] C++ **Unit tests**
  - [*] **integration tests**
- [*] :book: CHANGELOG entry made
- [*] Backports
  - [*] Backport for 3.10: This is it.

### Related

Original PR: https://github.com/arangodb/arangodb/pull/17981

### CODE REVIEW

File: arangod/RocksDBEngine/RocksDBGeoIndex.cpp [hunk 1]
```diff
@@ -759,9 +773,9 @@ std::unique_ptr<IndexIterator> RocksDBGeoIndex::iteratorForCondition(
   if (!params.sorted &&
       (params.filterType == geo::FilterType::CONTAINS ||
        params.filterType == geo::FilterType::INTERSECTS) &&
-      (params.minDistanceRad() < geo::kRadEps &&
-       params.maxDistanceRad() >
-           geo::kMaxRadiansBetweenPoints - geo::kRadEps)) {
+      (params.minDistance <= geo::kMetersEps &&
+       params.maxDistance >=
+           geo::kMaxDistanceBetweenPoints - geo::kMetersEps)) {
```
Comment (MBkkt) [inline arangod/RocksDBEngine/RocksDBGeoIndex.cpp:778]: https://github.com/arangodb/arangodb/pull/17981#discussion_r1085406286

**Review by MBkkt — APPROVED** [review]

File: lib/Geo/GeoParams.h [hunk 2]
```diff
@@ -47,6 +47,7 @@ constexpr double kMaxRadiansBetweenPoints = kPi + kRadEps;
 // constexpr double kEarthRadiusInMeters = (6378.137 * 1000);
 // Volumetric mean radius
 constexpr double kEarthRadiusInMeters = (6371.000 * 1000);
+constexpr double kMetersEps = kRadEps * kEarthRadiusInMeters;
```
Comment (MBkkt) [inline lib/Geo/GeoParams.h:50]: Can be removed?

File: CHANGELOG [hunk 3]
```diff
@@ -1,6 +1,9 @@
 v3.10.3 (XXXX-XX-XX)
 --------------------
 
+* Activate RDB_CoveringIterator and use it for some geo index queries.
+  This speeds up and simplifies geo queries which do not use GEO_DISTANCE.
```
Comment (MBkkt) [inline CHANGELOG:5]: NIT

"simplifies geo queries" with geo index

and "don't use GEO_DISTANCE or GEO_IN_RANGE"?

**Review by MBkkt — APPROVED** [review]
LGTM

File: lib/Geo/GeoParams.h (hunk 2 above)
Comment (neunhoef) [reply lib/Geo/GeoParams.h:50]: Yes, will do.

File: CHANGELOG (hunk 3 above)
Comment (neunhoef) [reply CHANGELOG:5]: As far as I see the geo index does not support `GEO_IN_RANGE` at all. I will add "with geo index".

Comment (MBkkt) [reply CHANGELOG:5]: Ok, thanks!

**Review by jsteemann — APPROVED** [review]
LGTM

**Pull request merged**